src/schedule/zephyr_dp_schedule.c

```diff
@@ -90,7 +239,18 @@ void scheduler_dp_ll_tick(void *receiver_data, enum notify_id event_type, void *
 							       mod->sinks,
 							       mod->num_of_sinks);
 			if (mod_ready) {
-				/* TODO: step 2 - caclulate deadlines */
+				int period_clock_ticks = pdata->period *
+							 CONFIG_SYS_CLOCK_TICKS_PER_SEC;
+				/* period is in us - convert to seconds in next step
+				 * or it always will be zero - its fixed point calculation
+				 */
+				period_clock_ticks /= 1000000;
```

Comment:
Are we able to calculate the ticks once ? it seems to be constant if period is constant ?

Reply:
good point, as Zephyr takes "number of machine cycles since now" as a deadline, it may be easy calculated at task creation time.